docs/type_names.md

```
@@ -0,0 +1,29 @@
+# Type Names
+
+There a four type of type names in **Elijjah**.
+
+1. Generic
+
+~~~~
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)